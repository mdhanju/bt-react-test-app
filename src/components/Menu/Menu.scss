nav {
    a {
        &.nav-link.active {
            border-bottom: 1px solid black;
        }
    }
}