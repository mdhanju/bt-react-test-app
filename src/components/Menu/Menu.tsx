import React from 'react'
import { Nav, Navbar, NavItem, NavLink, DropdownItem } from 'reactstrap'
import { NavLink as RRNavLink } from 'react-router-dom'

const Menu = (): React.ReactElement => {
    return (
        <React.Fragment>
            <Navbar light expand="md" className="mt-3">
                <Nav navbar className="mx-auto">
                    <NavItem>
                        <NavLink
                            exact
                            to="/bt-test-home"
                            tag={RRNavLink}
                            activeClassName="active"
                        >
                            <h3>Home</h3>
                        </NavLink>
                    </NavItem>
                    <NavItem>
                        <h2> - </h2>
                    </NavItem>
                    <NavItem>
                        <NavLink
                            exact
                            to="/addProfile"
                            tag={RRNavLink}
                            activeClassName="active"
                        >
                            <h3>Sample Form</h3>
                        </NavLink>
                    </NavItem>
                </Nav>
            </Navbar>

            <DropdownItem divider />
        </React.Fragment>
    )
}

export default Menu
