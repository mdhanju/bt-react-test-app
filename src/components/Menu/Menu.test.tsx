import React from 'react'

import Menu from './Menu'

describe('<Menu />', () => {
    it('should render properly without crashing', () => {
        const wrapper = shallow(<Menu />)
        expect(wrapper.exists()).toBe(true)
        expect(wrapper.getElements()).toMatchSnapshot()
    })
})
