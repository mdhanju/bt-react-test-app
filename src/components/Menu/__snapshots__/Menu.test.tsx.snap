// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should render properly without crashing 1`] = `
Array [
  <React.Fragment>
    <Navbar
      className="mt-3"
      expand="md"
      light={true}
      tag="nav"
    >
      <Nav
        className="mx-auto"
        navbar={true}
        tag="ul"
        vertical={false}
      >
        <NavItem
          tag="li"
        >
          <NavLink
            activeClassName="active"
            exact={true}
            tag={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "NavLink",
                "propTypes": Object {
                  "activeClassName": [Function],
                  "activeStyle": [Function],
                  "aria-current": [Function],
                  "className": [Function],
                  "exact": [Function],
                  "innerRef": [Function],
                  "isActive": [Function],
                  "location": [Function],
                  "onClick": [Function],
                  "replace": [Function],
                  "strict": [Function],
                  "style": [Function],
                  "target": [Function],
                  "to": [Function],
                },
                "render": [Function],
              }
            }
            to="/"
          >
            <h3>
              Home
            </h3>
          </NavLink>
        </NavItem>
        <NavItem
          tag="li"
        >
          <h2>
             - 
          </h2>
        </NavItem>
        <NavItem
          tag="li"
        >
          <NavLink
            activeClassName="active"
            exact={true}
            tag={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "NavLink",
                "propTypes": Object {
                  "activeClassName": [Function],
                  "activeStyle": [Function],
                  "aria-current": [Function],
                  "className": [Function],
                  "exact": [Function],
                  "innerRef": [Function],
                  "isActive": [Function],
                  "location": [Function],
                  "onClick": [Function],
                  "replace": [Function],
                  "strict": [Function],
                  "style": [Function],
                  "target": [Function],
                  "to": [Function],
                },
                "render": [Function],
              }
            }
            to="/sample"
          >
            <h3>
              Sample Form
            </h3>
          </NavLink>
        </NavItem>
      </Nav>
    </Navbar>
    <DropdownItem
      divider={true}
      tag="button"
      toggle={true}
    />
  </React.Fragment>,
]
`;
